### PR TITLE
[NVIDIA][test] Tests for flashinfer TRTLLM BF16 MoE

### PR DESCRIPTION
**Pull request by Linda-Stadter**
## Purpose
Adding tests for trtllm bf16 moe backend added in PR [[NVIDIA] [feat] Integrate flashinfer Trtllmgen bf16 moe #32954](https://github.com/vllm-project/vllm/pull/32954)

## Test Plan
- unit and integration test for the new moe backend
- unit tests for utility functions
- Changing E2E tests to use fi cutlass because E2E tests triggered an intermittent issue with flashinfer
## Test Result

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

**Review by pavanimajety — APPROVED** [review]
Thanks @Linda-Stadter, the tests look good to me

File: tests/quantization/test_blackwell_moe.py [hunk 1]
```diff
@@ -178,3 +178,11 @@ def test_gptoss_eager(monkeypatch: pytest.MonkeyPatch):
         hf_overrides=HF_OVERRIDE_TEXT,
         extra_args=["--enforce-eager"],
     )
+
+
+## Qwen3 Next ##
+
+
+def test_qwen3_next_bf16_moe_flashinfer_trtllm(monkeypatch: pytest.MonkeyPatch):
```
Comment (pavanimajety) [inline tests/quantization/test_blackwell_moe.py:186]: 80B model may be too big for CI, can we use a smaller model?

**Review by pavanimajety — CHANGES_REQUESTED** [review]

File: tests/quantization/test_blackwell_moe.py (hunk 1 above)
Comment (Linda-Stadter) [reply tests/quantization/test_blackwell_moe.py:186]: I am overwriting the numbers of layers here from the original 48 to just 4 layers. The total parameters should be then around ~7B. This should be in the same ballpark as the other models in this test file. Does that clear up you concerns?

Comment (pavanimajety) [reply tests/quantization/test_blackwell_moe.py:186]: thanks for the clarification, @Linda-Stadter! I didn't check the test setup, this looks good!

**Review by pavanimajety — APPROVED** [review]

File: tests/kernels/moe/test_moe.py [hunk 2]
```diff
@@ -1558,3 +1558,103 @@ def run(
...
+            x=a,
+            router_logits=router_logits,
+        )
+
+        # Compute torch baseline
+        w1_original = w1.clone()
+        w2_original = w2.clone()
+        baseline_output = torch_moe(a, w1_original, w2_original, router_logits, topk)
+
+    close = torch.isclose(trtllm_output, baseline_output, atol=1e-1, rtol=0.85)
```
Comment (bnellnm) [inline tests/kernels/moe/test_moe.py:1659]: Aren't these tolerances a bit large for bf16?

Comment (Linda-Stadter) [reply tests/kernels/moe/test_moe.py:1659]: I was also wondering about this. I got these numbers and the way of checking accuracy from the tests in flashinfer 
https://github.com/flashinfer-ai/flashinfer/blob/30bf78e1a05d18aaebd484aa6f3c1db8e4c0b1b4/tests/moe/test_trtllm_gen_fused_moe.py#L1398
and
https://github.com/flashinfer-ai/flashinfer/blob/30bf78e1a05d18aaebd484aa6f3c1db8e4c0b1b4/tests/moe/test_trtllm_gen_fused_moe.py#L1675
